RPIPs/RPIP-59.md

```diff
@@ -77,14 +77,15 @@ A node operator MUST take 2 actions to start a validator: `deposit` and `stake`
 - `stake` SHALL revert unless at least `scrub_period` time has passed since ETH was assigned to the validator, to allow for validating the prestake
 - If the beacon chain stake is invalid, the validator SHALL be scrubbed 
 - `stake` SHALL stake the remaining 31 ETH to the beacon chain to make a complete validator
-- If `stake` is not called within `time_before_dissolve` after the ETH was assigned, the validator SHALL be dissolved, returning the unstaked balance to the deposit pool
+- If `stake` is not called within `time_before_dissolve` after the ETH was assigned [can `stake` be called by anyone? Just curious if someone could stall the entire queue for 2 weeks at a time if they wanted to (maybe they'd have no reason to and it's nbd...] , the validator SHALL be dissolved, returning the unstaked balance to the deposit pool
```

Comment:
I think this might have been part of the 2tx versus 3tx discussion? Knoshua and Kane were talking about incentives to call stuff at one point, though that might have been around prestake.

Reply:
This doesn't stall the entire queue, just 31 ETH worth.

Up a bit higher it looks like this is an NO-specific action "A node operator MUST take 2 actions to start a validator: `deposit` and `stake`". I think this makes sense.

Reply:
Is that 31 ETH worth per validator deposit (which is only 1.5ETH?), So if someone did: say 100 deposits of 1.5 ETH, if they wanted to they could stall 3100 ETH for two weeks? (Or even worse 31 ETH per 2 weeks, so 100*2 weeks?)

That's probably not how it works but just wanted to check

Reply:
Can stall 3100 ETH for two weeks in this scenario. At the cost of making your own ETH idle and paying a buncha gas.

Reply:
So the cost to stall current DP size for two weeks requires putting up (`(29,112/31)*1.5 ~= 1400 ETH`) worth of idle capital (and paying gas). I agree I don't see any incentive to do this necessarily, but should it be accounted for in security considerations? Just a (maybe outlandish) example: a competitor wants to stall rETH minting as a griefing vector, they continue to cycle 1400 ETH worth which stalls our current DP worth of rETH deposits for 2 weeks at a time each round they do it (hurts rETH APY and stalls growth)... If they had another 1400 ETH on standby they could just continue to have a constant two week stall that keeps cycling? (Maybe other deposits can get in line in between each cycle, but maybe no one wants to since NO queue is too long, especially with more ETH?) Express queue tickets could help but is limited in number.

Maybe this is already listed as a consideration:
> Disabling social assignments increases the likelihood of ETH accumulating in the deposit pool while validators are waiting in the queue. But the node operators in the queue that would be assigned ETH have a direct incentive to execute the assignments. In addition, we recommend that the pDAO funds (for example through the GMC) development and execution of assignment bots that assign ETH in the deposit pool to the queue at a reasonable gas price.

Or maybe this isn't a valid concern/too high of a cost to do this, with not enough incentive?

Reply:
I don't see any harm in including it as a security consideration? I don't feel very strongly either way though.

Reply:
I think the consideration you highlighted is different.

> If a validator is dissolved the bonded value SHALL be recoverable. This MAY require further action from the node operator. This MAY temporarily require additional ETH from the node operator.
The above is the relevant text. 

You're welcome to add a security consideration bullet. I think after one use folks would get in line and it would just be an annoyance. But nonetheless:
> There is potential for moderate griefing by creating n validators and never calling `stake`. The griefer keeps n*31 protocol ETH idle for 2 weeks at the cost of keeping n of their own ETH idle and paying gas to `deposit` and reclaim funds from `dissolve`. The text notes that reclaiming funds MAY require further action from the node operator and MAY require additional temporary ETH. Friction could be added in either of those steps should this variety of griefing prove problematic. Alternatively, a small fee to reclaim funds would also be effective (though that would require a new specification etc.).

Reply:
:rocket: